### ISSUE TITLE
[iOS] `TextBox` keyboard "Done" button

### ISSUE THREAD

**Issue by mikernet** [unoplatform/uno#7247]
## What would you like to be added:

The ability to add a "Done" button to the toolbar above the keyboard that dismisses the keyboard:

![image](https://user-images.githubusercontent.com/1700111/135388882-bcbf599e-2ec8-45a4-96d7-04f6b3f6aa21.png)

Not sure if this is currently possible but I couldn't find anything on the topic.

## Why is this needed:

A mechanism to easily dismiss the keyboard is especially important on multi-line TextBoxes since the "Return" keyboard button does not dismiss the keyboard and the TextBox may take up most or all of the remaining screen space.

This would also allow using the "Return" key to move to the next input field on non-multiline textboxes while still providing a way to dismiss the keyboard. ~~Speaking of which - is there a way to change the text for the return button to "Next" (i.e. supporting iOS's `returnKeyType` setting [similar to this](https://docs.microsoft.com/en-us/dotnet/api/uikit.uireturnkeytype?view=xamarin-ios-sdk-12))?~~ (resolved in comments below)

## For which Platform:

- [X] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

**Comment by Youssef1313** [comment]
It looks like there is an API for iOS in Uno to control that:

https://github.com/unoplatform/uno/blob/45162f3aac50c8a232eaf0f25006f77e95bc6b5e/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.iOS.cs#L274

Not sure if this is what you want?

**Comment by mikernet** [comment]
That is part of what I want, yes, thank you. The other part is the `Done` button in the toolbar above the keyboard. Is there an API for that as well? Is there a way to discover these features without having to browse the source?

**Comment by MartinZikmund** [comment]
Yes, this specific feature is not supported directly from Uno yet. You may be able to use the approach described in this Stack Overflow answer within `#if __IOS__` block, but it is definitely something worth adding into Uno as an extension/helper

**Comment by MartinZikmund** [comment]
This was implemented via the `InputReturnType` feature on all targets - https://platform.uno/docs/articles/controls/[TextBox](https://platform.uno/docs/articles/controls/TextBox.html#customizing-the-enter-key-appearance).html#customizing-the-enter-key-appearance

**Comment by sasakrsmanovic** [comment]
@MartinZikmund  - Mike couldn't add to this issue so we started a new one --> https://github.com/unoplatform/uno/issues/24384
